### PR TITLE
Populate Collection.search_vector search index

### PR DESCRIPTION
**Pull request by cutwater**
Add a migration that generates full text search index
for existing collections.

Closes: #1525

### CODE REVIEW

File: galaxy/main/migrations/0142_collection_search.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  WHERE cv.id = c.latest_version_id
+)
+WHERE c.latest_version_id IS NOT NULL
+'''
+
+CREATE_COLLECTIONS_TS_VECTOR_TRIGGER = f'''
+CREATE OR REPLACE FUNCTION update_collection_ts_vector()
+RETURNS TRIGGER AS $$
+BEGIN
+  IF new.latest_version_id IS NOT NULL THEN
```
Comment (ironfroggy) [inline galaxy/main/migrations/0142_collection_search.py]: `new` is uppercase everywhere else. Is that significant?

File: galaxy/main/migrations/0142_collection_search.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+      SELECT
+        setweight(to_tsvector(
+          coalesce(string_agg(cvc ->> 'name', ' '), '')
+        ), 'C')
+      FROM jsonb_array_elements(cv.contents) AS cvc
+    )
+    || setweight(to_tsvector(
+         coalesce(cv.metadata ->> 'description', '')
+       ), 'D')
+'''
```
Comment (ironfroggy) [inline galaxy/main/migrations/0142_collection_search.py]: Please document what this query will be doing.

Comment (cutwater) [reply galaxy/main/migrations/0142_collection_search.py]: Will do.

**Comment by cutwater** [comment]
Closing to prevent accidental merge until fixed.

**Pull request merged**